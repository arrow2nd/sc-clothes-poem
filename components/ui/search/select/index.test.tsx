import { fireEvent, render } from '@testing-library/react'
import { act } from 'react-dom/test-utils'

import Select from './index'

describe('Select', () => {
  const props = {
    placeholder: 'placeholder',
    options: ['opt1', 'opt2', 'opt3']
  }

  test('プレースホルダが反映されているか', () => {
    const { container } = render(<Select {...props} onChange={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })

  test('文字列を入力して要素を選択できるか', async () => {
    const mock = jest.fn()
    const { container, getByRole } = render(
      <Select {...props} onChange={mock} />
    )

    await act(async () => {
      const combobox = getByRole('combobox')

      fireEvent.change(combobox, {
        target: { value: '1' }
      })

      // ドロップダウンが開くまで待機
      await new Promise((r) => setTimeout(r, 100))

      fireEvent.keyDown(combobox, {
        key: 'Enter',
        code: 'Enter',
        charCode: 13
      })
    })

    expect(container).toMatchSnapshot()
  })
})
