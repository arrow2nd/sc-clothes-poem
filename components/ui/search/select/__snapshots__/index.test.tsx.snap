// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select プレースホルダが反映されているか 1`] = `
<div>
  <div
    class="w-full lg:w-72 m-2 bg-white css-b62m3t-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-placeholder-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class=" css-1ycg57x-control"
    >
      <div
        class=" css-319lph-ValueContainer"
      >
        <div
          class=" css-1yaegve-placeholder"
          id="react-select-placeholder-placeholder"
        >
          <p>
            placeholder
          </p>
        </div>
        <div
          class=" css-17k8k4u-Input"
          data-value=""
        >
          <input
            aria-autocomplete="list"
            aria-controls="react-select-placeholder-listbox"
            aria-describedby="react-select-placeholder-placeholder"
            aria-expanded="false"
            aria-haspopup="true"
            aria-owns="react-select-placeholder-listbox"
            autocapitalize="none"
            autocomplete="off"
            autocorrect="off"
            class=""
            id="react-select-placeholder-input"
            role="combobox"
            spellcheck="false"
            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
            tabindex="0"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class=" css-1hb7zxy-IndicatorsContainer"
      >
        <span
          class=" css-p70wra-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          class=" css-lp1szy-indicatorContainer"
        >
          <svg
            aria-hidden="true"
            class="css-tj5bde-Svg"
            focusable="false"
            height="20"
            viewBox="0 0 20 20"
            width="20"
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select 文字列を入力して要素を選択できるか 1`] = `
<div>
  <div
    class="w-full lg:w-72 m-2 bg-white css-b62m3t-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-placeholder-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class=" css-1ycg57x-control"
    >
      <div
        class=" css-319lph-ValueContainer"
      >
        <div
          class=" css-354nsk-singleValue"
        >
          opt1
        </div>
        <div
          class=" css-17k8k4u-Input"
          data-value=""
        >
          <input
            aria-autocomplete="list"
            aria-controls="react-select-placeholder-listbox"
            aria-expanded="false"
            aria-haspopup="true"
            aria-owns="react-select-placeholder-listbox"
            autocapitalize="none"
            autocomplete="off"
            autocorrect="off"
            class=""
            id="react-select-placeholder-input"
            role="combobox"
            spellcheck="false"
            style="opacity: 0; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
            tabindex="0"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class=" css-1hb7zxy-IndicatorsContainer"
      >
        <span
          class=" css-p70wra-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          class=" css-lp1szy-indicatorContainer"
        >
          <svg
            aria-hidden="true"
            class="css-tj5bde-Svg"
            focusable="false"
            height="20"
            viewBox="0 0 20 20"
            width="20"
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
